python/cudf/cudf/tests/test_json.py

```diff
@@ -345,7 +347,9 @@ def test_json_bool_values():
     # boolean values should be converted to 0/1
     np.testing.assert_array_equal(pd_df[1], cu_df["1"].to_numpy())
 
-    cu_df = cudf.read_json(buffer, lines=True, dtype=["bool", "long"])
+    cu_df = cudf.read_json(
+        buffer, lines=True, dtype={"0": "bool", "1": "long"}
+    )
```

Comment:
Is the `dtype` argument going to be a breaking change for the `engine='cudf'` reader?

Reply:
Nope, `dtype` is going to continue to work for `engine='cudf'` reader. But the reason I've deprecated supporting the list of dtypes in `dtype` param is there is no way we can give the column a name with `schema_info` being introduced going forward.

Reply:
`dtype=list` is going to a breaking change once we drop it. But this PR is just deprecating.